debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/DumpModuleCommand.java

```diff
@@ -0,0 +1,240 @@
+			}
+			break;
+		default:
+			break; /* default stay status.HELP */
+		}
+		try {
+			int result;
+			J9ModulePointer modulePtr = J9ModulePointer.cast(Long.decode(filterArg));
+			J9ClassLoaderPointer classLoaderPtr = modulePtr.classLoader();;
+			switch (status) {
+			case CLASS:
+				result = ModularityHelper.iterateClassLoaderClasses(out, classFilter, classOutput, filterArg, classLoaderPtr);
+				System.out.printf("Found %d class%s\n", result, (1 == result ? "" : "es"));
+				break;
+			case MODULE:
+				result = ModularityHelper.iterateModules(out, moduleFilter, moduleOutputter, filterArg);
+				System.out.printf("Found %d module%s\n", result, (1 == result ? "" : "s"));
+				break;
+			case PACKAGE:
+				result = ModularityHelper.iterateClassLoaderPackages(out, packageFilter, packageOutputter, filterArg, classLoaderPtr);
+				System.out.printf("Found %d package%s\n", result, (1 == result ? "" : "s"));
+				break;
+			case ALL:
+				out.printf("Module:\n");
+				ModularityHelper.printJ9Module(modulePtr, out);
+				/* Run requires */
+				out.printf("Requires:\n");
+				moduleFilter = DumpModuleCommand::filterModuleRequired;
+				moduleOutputter = ModularityHelper::printJ9Module;
+				result = ModularityHelper.iterateModules(out, moduleFilter, moduleOutputter, filterArg);
+				System.out.printf("Found %d required module%s\n", result, (1 == result ? "" : "s"));
+				/* Run exports */
+				out.printf("Exports:\n");
+				packageFilter = DumpModuleCommand::filterPackageModuleAndExport;
+				packageOutputter = ModularityHelper::printPackageExportVerbose;
+				result = ModularityHelper.iteratePackages(out, packageFilter, packageOutputter, filterArg);
+				System.out.printf("Found %d exported package%s\n", result, (1 == result ? "" : "s"));
+				break;
+			default:
+				printHelp(out);
+				break;
+			}
+		} catch (CorruptDataException e) {
+			throw new DDRInteractiveCommandException(e);
+		}
+
+		return;
+	}
+
+	private static boolean filterClassByModule(J9ClassPointer classPtr, String targetModuleAddress) throws CorruptDataException {
+		String moduleAddress = classPtr.module().getHexAddress();
+		return targetModuleAddress.equalsIgnoreCase(moduleAddress);
+	}
+
+	private static boolean filterModuleRequired(J9ModulePointer modulePtr, String targetModuleAddress) throws CorruptDataException {
+		boolean result = false;
+		J9HashTablePointer readTable = modulePtr.readAccessHashTable();
+		HashTable<J9ModulePointer> readModuleHashTable = ModuleHashTable.fromJ9HashTable(readTable);
+		SlotIterator<J9ModulePointer> readSlotIterator = readModuleHashTable.iterator();
+		while (readSlotIterator.hasNext()) {
+			J9ModulePointer readModulePtr = readSlotIterator.next();
+			String readHexAddress = readModulePtr.getHexAddress();
+			if (targetModuleAddress.equals(readHexAddress)) {
+				result = true;
+				break;
+			}
+		}
+		return result;
+	}
+
+	/**
+	 * Check whether a givin class is exported. Both
+	 * `export <class>` and `export <class> to
```

Comment:
Classes are not exported, but packages are. Please fix the comments to match the code.